radixdlt/src/main/java/com/radixdlt/middleware2/store/CommittedAtomsStore.java

```diff
@@ -77,30 +84,42 @@ public CommittedAtomsStore(
 	) {
 		this.committedAtomSender = Objects.requireNonNull(committedAtomSender);
 		this.store = Objects.requireNonNull(store);
+		this.persistentVertexStore = Objects.requireNonNull(persistentVertexStore);
 		this.commandToBinaryConverter = Objects.requireNonNull(commandToBinaryConverter);
 		this.clientAtomToBinaryConverter = Objects.requireNonNull(clientAtomToBinaryConverter);
 		this.atomIndexer = Objects.requireNonNull(atomIndexer);
 		this.serialization = Objects.requireNonNull(serialization);
 		this.hasher = hasher;
 	}
 
-	private Optional<ClientAtom> getAtomByParticle(Particle particle, boolean isInput) {
+	private boolean getAtomByParticle(Particle particle, boolean isInput) {
```

Comment:
Probably method name should be changed, as there is no `get` operation anymore.

Reply:
good point